### PR TITLE
Improved fix for uncontrollable dive

### PR DESCRIPTION
**Pull request by donstim**
Improves upon the previous PR #2698 

## Summary of Changes

- Improved notice of Flaps auto-retraction from config 1 + F to config 1 in Upper ECAM and Core Speeds
- Enabled auto-retraction in config for drag, which also enables the flap retraction in the eternal view

Allows the auto-retraction of 1+F to 1 for drag, but not for lift. This enables the animation in the external model to show the automatic flap retraction. So, now we can have the visual flap retraction (and the drag reduction associated with it), but retains the lift of 1+F just like the preliminary fix. The external view will show the flap retract above 210 knots, but it will not lead to the unrecoverable stall condition.
## Additional context

Discord username (if different from GitHub): donbikes#4084

## Testing instructions

This should be tested in a similar manner as the testing for PR #2698, except add a check of the external view that the flap retracts at speeds above 210 knots, and of the internal view that the flap symbol also shows the flap retraction.

1. Start on the runway (any runway should do) at the default weight.
2. Go to external view. No need to be distracted by any instruments, warnings,, CDU setup, or anything else. 
3. Release parking brakes, go to TOGA thrust, rotate around 130 knots (not critical), and retract the landing gear with positive rate of climb.
4. Just keep a gentle climb with the primary purpose of allowing the airplane to accelerate. Do not retract the flap.
5. When accelerating through 210 knots, confirm that the flap auto-retracts. (It takes a little while, and depending on how fast you are accelerating it may not appear to start right away.)
6. Switch to internal view and confirm that the ECAM flap symbol shows that the flap has retracted (from 1+F to 1) and that VFE on the PFD speed tape has increased from 215 knots to 230 knots.
7. Reduce speed to below 210 knots and confirm that the flap re-deploys in the external view, and that the ECAM symbol changes back to 1+F, and VFE reduces again to 215 knots.
8. Increase speed again, and either disable the overspeed aural warning by pushing the master caution button or switch back to external view and continue to accelerate out to 280-290 knots to verify that this PR does not cause the unrecoverable stall issue to reappear.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link

### CODE REVIEW

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -18,6 +18,9 @@
 1. [PFD] Fixed wrong Vfe and Vfen indication - @MisterChocker (Leon)
 1. [SOUND] Reworked interior and exterior apu sound - @hotshotp (Boris#9134)
 1. [SOUND] Added new MCDU button sounds - @hotshotp (Boris#9134)
+1. [FLIGHTMODEL] Improved fix for flaps 1 unrecoverable dives - @donstim (donbikes#4084)
+1. [ECAM] Improved flap auto-retraction detection - @MisterChocker (Leon)
+1. [CORE] Improved flap auto-retraction detection - @MisterChocker (Leon)
```
[the remark this answers is not in the document]
Comment (donstim) [reply .github/CHANGELOG.md:24]: They are different files (different sections). I can combine into one as ECAM/CORE if that would be better.

[the remark this answers is not in the document]
Comment (pepperoni505) [reply .github/CHANGELOG.md:24]: Yes, please do that

[the remark this answers is not in the document]
Comment (donstim) [reply .github/CHANGELOG.md:24]: Okay, will do!

**Review by MisterChocker — APPROVED** [review]
LGTM

**Comment by NickFarin** [comment]
Category: QA Trial
Name: Discord User: Nick F[Z+8] #5327
Date of testing: Current date 02/012021
Version of the sim: 1.12.13.0
PR Tested: #2804 
Object of test: Improved fix for uncontrollable dive and automatic flap retraction and extension
Testing technique: Spawned in at the runway at KMCO and took off and observed flaps
Steps you've made to test the Pull Request:
1. Spawned in at the active runway at KMCO and applied TOGA power
2. Climbed out and moved to wing view
3. Observed if flaps retracted at 210 knots
4. As soon as the flaps started moving, I moved into the cockpit view to see if the speed tape had moved to 230 knots and checked if ECAM shows flaps 1
5. After passing 230 knots, I retarded the thrust levers and slowed down
6. Moved into wing view to see if flaps extended when passing 210 knots
7.  Moved back to cockpit view to see if ECAM shows flaps as 1+F
8.  Moved thrust levers into TOGA and accelerated to 290 knots
9. Checked if there was an uncontrollable dive after passing 290 knots

Flight notes (If required):

Issues (If Any): 
Overall Rating: Good
Conclusions of the testing: Test of the fix for uncontrollable dive and automatic flap retraction and extension went normal and no issues were found.

**Comment by ImenesFBW** [comment]
Name: Imenes#8739
Date of testing: Current date 06/01/2021
Version of the sim: 1.12.13.0
PR Tested: #2804
Object of test: Checking flap retraction 
Testing technique: Try-out (multiple tests)

## Steps you've made to test the Pull Request:

 * Spawned on the runway
 * Switched to external view 
 * Opened the applicable instruments in a window 
 * T/O gear up accel over 210 knots in 1+F to confirm auto retraction
 * Speed reduction below 210 to confirm re-deployment 
 * Speed increase to 285 kts to check for the stall issue
 
### Flight notes:

 * Fleps retract over 210 knots and redeploy under 210 knots
 * Speedtape adjusts between 215 and 230 knots correctly given the config.
 * The stall issue at high-speeds is not present.
 * Pictures included below 

## Overall Rating: Needs improvement

#### Issues:

 * There is one slight issue I have noticed in my test, the ECAM flap symbol does not adjust correctly. The left side stays deployed until you manually retract the flaps.
 * [Video of one of the tests here](https://streamable.com/3kfn9j)
 
#### Conclusions of the testing:

 * Most of the functionality is there and working as documented, except the issue with the ECAM flap symbol, described above.

#### Pictures:

DB1

DB2

DB3

**Comment by donstim** [comment]
> 
> ## Overall Rating: Needs improvement
> #### Issues:
> * There is one slight issue I have noticed in my test, the ECAM flap symbol does not adjust correctly. The left side stays deployed until you manually retract the flaps.
> * [Video of one of the tests here](https://streamable.com/3kfn9j)
> 
> #### Conclusions of the testing:
> * Most of the functionality is there and working as documented, except the issue with the ECAM flap symbol, described above.
> 
That's exactly how it is supposed to work. The flap retracts, but not the slat. So you start at CONF 1+F, but after the auto-retraction you end up with CONF 1.

Nice video by the way!  If you have done a view from the side or the front, you would have seen that the slat remains deployed. You can just barely see the edge of it being separated from the wing at times in your video from the rear.

You really should have waited until they plowed that runway!

**Comment by ImenesFBW** [comment]
> > ## Overall Rating: Needs improvement
> > #### Issues:
> > 
> > * There is one slight issue I have noticed in my test, the ECAM flap symbol does not adjust correctly. The left side stays deployed until you manually retract the flaps.
> > * [Video of one of the tests here](https://streamable.com/3kfn9j)
> > 
> > #### Conclusions of the testing:
> > 
> > * Most of the functionality is there and working as documented, except the issue with the ECAM flap symbol, described above.
> 
> That's exactly how it is supposed to work. The flap retracts, but not the slat. So you start at CONF 1+F, but after the auto-retraction you end up with CONF 1.
> 
> Nice video by the way! If you have done a view from the side or the front, you would have seen that the slat remains deployed. You can just barely see the edge of it being separated from the wing at times in your video from the rear.
> 
> You really should have waited until they plowed that runway!


Sorry for some reason I was reading it as FLAP right and left instead of S and F, error on my part changing the review to GOOD.

**Comment by Awemeter** [comment]
Category: QA Tester
Name: Awemeter#7919
Date of testing: 1/5/2021
Version of the sim: 1.12.13.0
PR Tested: #2804 
Object of test: Flap and Slats
Testing technique: Followed testing instructions
Steps you've made to test the Pull Request:
1. Spawned on runway 29 at LOWW
2. Open window instruments in external view.
3. Proceeded to go toga and rotated around 130kts
4. Passed 215 knots and flaps auto retracted and vfe tape got higher
5. Reduced to 130kts and flaps extended  and vfe tape go lower
6. increased to 280kts and flaps retracted and vfe tape got higher. 

Flight notes (If required):
Did not suffer uncontrollable dive

Issues (If Any): N/A
Overall Rating: Good
Conclusions of the testing: Pr works properly and did not find any issues. Thanks for your hard work

**Pull request merged**